.github/workflows/build_and_test.yml

```diff
@@ -227,6 +227,20 @@ jobs:
         luarocks --version
         luarocks config
 
+    - name: Generate test rerun filename
+      run: |
+           echo FAILED_TEST_FILES_FILE=$(echo '${{ matrix.suite }}-${{ matrix.split }}' | tr A-Z a-z | sed -Ee 's/[^a-z0-9]+/-/g').txt >> $GITHUB_ENV
+
+
+    - name: Download test rerun information
+      uses: actions/download-artifact@v3
+      continue-on-error: true
+      with:
+        name: ${{ env.FAILED_TEST_FILES_FILE }}
```

Comment:
I can see the name is generated from the test suite/split, but is this artifact also scoped to a particular branch/ref/PR in some way that I'm not seeing?

What happens when pull-request A has some failures, and then pull request B is opened a few minutes later? What stops pull request B from retrieving the failed list from pull request A and only running a subset of the tests?

Reply:
The failing tests file is now scoped to the workflow run.